libsolidity/formal/CHC.cpp

```diff
@@ -40,21 +40,20 @@ CHC::CHC(
 	smt::SMTSolverChoice _enabledSolvers
 ):
 	SMTEncoder(_context),
-#ifdef HAVE_Z3
-	m_interface(
-		_enabledSolvers.z3 ?
-		dynamic_pointer_cast<smt::CHCSolverInterface>(make_shared<smt::Z3CHCInterface>()) :
-		dynamic_pointer_cast<smt::CHCSolverInterface>(make_shared<smt::CHCSmtLib2Interface>(_smtlib2Responses, _smtCallback))
-	),
-#else
-	m_interface(make_shared<smt::CHCSmtLib2Interface>(_smtlib2Responses, _smtCallback)),
-#endif
 	m_outerErrorReporter(_errorReporter),
 	m_enabledSolvers(_enabledSolvers)
 {
-	(void)_smtlib2Responses;
 	(void)_enabledSolvers;
```

Comment:
Unrelated to this PR, but for this it may be worth to look at https://en.cppreference.com/w/cpp/language/attributes/maybe_unused and use that instead.

Reply:
done